lib/cover-project.nix

```diff
@@ -0,0 +1,84 @@
+{ pkgs, lib, haskellLib }:
+
+# List of project packages to generate a coverage report for
+{ packages
+, coverageReportOverrides ? {}
+}:
+
+let
+  getPackageCoverageReport = packageName: (coverageReportOverrides."${packageName}" or packages."${packageName}".coverageReport);
+
+  # Create table rows for an project coverage index page that look something like:
+  #
+  # | Package          |
+  # |------------------|
+  # | cardano-shell    |
+  # | cardano-launcher |
+  packageTableRows = package: with lib;
+    let
+      testsOnly = filterAttrs (n: d: isDerivation d) package.components.tests;
+      testNames = mapAttrsToList (testName: _: testName) testsOnly;
+    in
+      concatStringsSep "\n" (map (testName:
+      ''
+      <tr>
+        <td>
+          <a href="${package.identifier.name}-${package.identifier.version}/hpc_index.html">${package.identifier.name}</href>
+        </td>
+      </tr>
+      '') testNames);
+
+  projectIndexHtml = pkgs.writeText "index.html" ''
+  <html>
+    <head>
+      <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
+    </head>
+    <body>
+      <table border="1" width="100%">
+        <tbody>
+          <tr>
+            <th>Package</th>
+          </tr>
+
+          ${with lib; concatStringsSep "\n" (mapAttrsToList (_ : packageTableRows) packages)}
+
+        </tbody>
+      </table>
+    </body>
+  </html>
+  '';
+in pkgs.runCommand "project-coverage-report"
+  { buildInputs = (with pkgs; [ghc]); }
```

Comment:
Same comment about choice of GHC here.

Reply:
Fixed: https://github.com/input-output-hk/haskell.nix/pull/762/files#diff-8ee3808821db3b9f819360e568c5004aR51